index.js

```diff
@@ -22,69 +21,104 @@ function writeFile (filename, data, options, callback) {
     options = null
   }
   if (!options) options = {}
-  fs.realpath(filename, function (_, realname) {
-    _writeFile(realname || filename, data, options, callback)
-  })
-}
-function _writeFile (filename, data, options, callback) {
-  var tmpfile = getTmpname(filename)
 
-  if (options.mode && options.chown) {
-    return thenWriteFile()
-  } else {
-    // Either mode or chown is not explicitly set
-    // Default behavior is to copy it from original file
-    return fs.stat(filename, function (err, stats) {
-      if (err || !stats) return thenWriteFile()
+  var truename
+  var fd
+  var tmpfile
 
-      options = Object.assign({}, options)
-      if (!options.mode) {
-        options.mode = stats.mode
-      }
-      if (!options.chown && process.getuid) {
-        options.chown = { uid: stats.uid, gid: stats.gid }
-      }
-      return thenWriteFile()
+  new Promise(function (resolve) {
```

Comment:
Yes, but that isn't automatically transitive to libraries it uses. Each needs its own major bump. (Believe me, I've tried to slip those things in before, it breaks people and makes them sad. =D)

But it's not a big deal, npm can take write-file-atomic@3 just fine.